File: elearning-backend-project2/src/main/java/com/elearning/web/service/StudentService.java
package com.elearning.web.service;


import java.util.List;

import com.elearning.web.model.Assignment;
import com.elearning.web.model.Course;
import com.elearning.web.model.Student;
import com.elearning.web.model.Teacher;


public interface StudentService {
	
	public Student addStudent(Student student);
	public List<Course> getCourses();
	public Student enrollCourse(Integer id,Integer sid);
	public Student login(String userId, String password);
	//public List<Student> getStudent();
	//public List<Course> getenrolledcourses();
 
	
	
	
	
	
	
	public Assignment addAssignment(Assignment assignment);
	public Assignment getOneAssignment(Integer assignment_id);
	public Assignment updateAssignment(Assignment assignment);
	public Assignment deleteAssignment(Integer assignment_id);


}

File: elearning-frontend-project2/src/app/student.service.ts
import { HttpHeaders } from '@angular/common/http';
import { HttpParams } from '@angular/common/http';
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';

@Injectable({ providedIn: 'root' })
export class StudentService {
  readonly baseUrl = 'https://elearning-application-revature.herokuapp.com/';
  headers = new HttpHeaders().append('Content-Type', 'application/json');
  constructor(private httpClient: HttpClient) {}
  addStudent(student): Observable<any> {
    return this.httpClient.post(this.baseUrl + 'add-student', student, {
      headers: this.headers
    });
  }
  getCourses(): Observable<any> {
    return this.httpClient.get(this.baseUrl + 'get-courses');
  }

  public login(userid: string, password: string): Observable<any> {
    return this.httpClient.post(this.baseUrl + 'student-login', null, {
      params: new HttpParams()
        .append('userid', userid)
        .append('password', password)
    });
  }

  public enrollCourse(courseid, studentid) {
    return this.httpClient.post(this.baseUrl + 'enroll-course', null, {
      params: new HttpParams().append('cid', courseid).append('sid', studentid)
    });
  }
}
